src/components/themed/FlipInput.js

```diff
@@ -373,11 +374,20 @@ class FlipInputComponent extends React.PureComponent<Props, State> {
     }
   }
 
+  handleOnError(error?: string) {
+    if (this.props.onError != null) {
+      this.props.onError(error)
+    }
+  }
+
   onKeyPress(keyPressed: string, decimalAmount: string, maxEntryDecimals: number, setAmounts: (Props, string) => Amounts) {
     keyPressed = keyPressed.replace(',', '.')
     if (keyPressed === 'Backspace') {
       this.setStateAmounts(truncateDecimals(decimalAmount.slice(0, -1), maxEntryDecimals), setAmounts)
-    } else if (checkKeyPress(keyPressed, decimalAmount)) {
+    } else if (!checkKeyPress(keyPressed, decimalAmount)) {
```

Comment:
Instead of flipping the condition, keep it the same and then your else would be the error.